components/de_amp/browser/de_amp_throttle.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_DE_AMP_BROWSER_DE_AMP_THROTTLE_H_
+#define BRAVE_COMPONENTS_DE_AMP_BROWSER_DE_AMP_THROTTLE_H_
+
+#include <memory>
+
+#include "base/memory/scoped_refptr.h"
+#include "base/task/sequenced_task_runner.h"
+#include "brave/components/body_sniffer/body_sniffer_throttle.h"
+#include "content/public/browser/web_contents.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/mojom/url_response_head.mojom-forward.h"
+#include "url/gurl.h"
+
+namespace de_amp {
+
+// Throttle for AMP HTML detection.
+// If AMP page, redirect request to non-AMP canonical link.
```

Comment:
`If AMP page, cancel the request and initiate a new one to non-AMP canonical link.` ?